xbmc/video/jobs/VideoLibraryCleaningJob.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ *      Copyright (C) 2014 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "VideoLibraryCleaningJob.h"
+#include "dialogs/GUIDialogExtendedProgressBar.h"
+#include "video/VideoDatabase.h"
+
+using namespace std;
+
+CVideoLibraryCleaningJob::CVideoLibraryCleaningJob(const std::set<int>& paths /* = std::set<int>() */, bool showDialog /* = false */)
+  : CVideoLibraryProgressJob(NULL),
+    m_paths(paths),
+    m_showDialog(showDialog)
+{ }
+
+CVideoLibraryCleaningJob::CVideoLibraryCleaningJob(const std::set<int>& paths, CGUIDialogProgressBarHandle* progressBar)
+  : CVideoLibraryProgressJob(progressBar),
+    m_paths(paths),
+    m_showDialog(false)
+{ }
+
+CVideoLibraryCleaningJob::~CVideoLibraryCleaningJob()
+{ }
+
+bool CVideoLibraryCleaningJob::operator==(const CJob* job) const
```

Comment:
Oh, yeah... TIL: that actually works.
I think this could be problematic then (depends on when job manager checks that i guess). For instance: queuing a new scan while another one is running should really queue a new one as changes occurring since may not be picked up by the scan that have progressed.